.github/workflows/ci.yml

```
@@ -21,6 +21,10 @@ jobs:
     - uses: subosito/flutter-action@4389e6cbc6cb8a4b18c628ff96ff90be0e926aa8
       with:
         flutter-version: ${{ matrix.flutter_version }}
+    - name: Remove additional NDKs (Android Gradle plugin 3.5.0 workaround)
+      run: |
+        chmod +x .github/apply-ndk-workaround &&
+        .github/apply-ndk-workaround
```

If you don't set +x in the repository, maybe it's worth considering:
```suggestion
        sh .github/apply-ndk-workaround
```

Or `bash` if it is really needed.

---

The problem with this is that you need to know which shell to run in the action's yaml, but you also need the script itself to document which shell syntax it's using, which is something the shebang in the script already accomplishes. 

By leaving the shebang in the script and using +x here we avoid duplicating the shell information.

---

> The problem with this is that you need to know which shell to run in the action's yaml,

I don't think knowing the environment your code runs in is a problem.

What's the reason to use `bash` instead of `sh` in this case?

PS. I'm fine with executing the script in the way you proposed. However, I don't see the duplication between shebang and running shell explicitly. One does not interfere with another and can be used for different approaches. Same with file extensions which we don't consider redundant to shebangs.